src/annotator/util/test/buckets-test.js

```diff
@@ -0,0 +1,275 @@
+      stubbedInnerHeight = sinon.stub(window, 'innerHeight').value(410);
+    });
+
+    afterEach(() => {
+      stubbedInnerHeight.restore();
+    });
+
+    it('returns an Array of anchors that are offscreen above', () => {
+      const positionPoints = constructPositionPoints(fakeAnchors);
+
+      assert.deepEqual(positionPoints.above, [fakeAnchors[0], fakeAnchors[1]]);
+    });
+
+    it('returns an Array of anchors that are offscreen below', () => {
+      const positionPoints = constructPositionPoints(fakeAnchors);
+
+      assert.deepEqual(positionPoints.below, [fakeAnchors[4], fakeAnchors[5]]);
+    });
+
+    it('does not return duplicate anchors', () => {
+      const positionPoints = constructPositionPoints([
+        fakeAnchors[0],
+        fakeAnchors[0],
+        fakeAnchors[5],
+        fakeAnchors[5],
+      ]);
+
+      assert.deepEqual(positionPoints.above, [fakeAnchors[0]]);
+      assert.deepEqual(positionPoints.below, [fakeAnchors[5]]);
+    });
+
+    it('returns an Array of position points for on-screen anchors', () => {
+      const positionPoints = constructPositionPoints(fakeAnchors);
+
+      // It should return two "point" positions for each on-screen anchor,
+      // one representing the top of the anchor's highlight box, one representing
+      // the bottom position
+      assert.equal(positionPoints.points.length, 4);
+      // The top position of the first on-screen anchor
+      assert.deepEqual(positionPoints.points[0], [201, 1, fakeAnchors[2]]);
+      // The bottom position of the first on-screen anchor
+      assert.deepEqual(positionPoints.points[1], [251, -1, fakeAnchors[2]]);
+      // The top position of the second on-screen anchor
+      assert.deepEqual(positionPoints.points[2], [301, 1, fakeAnchors[3]]);
+      // The bottom position of the second on-screen anchor
+      assert.deepEqual(positionPoints.points[3], [351, -1, fakeAnchors[3]]);
+    });
+
+    it('sorts on-screen points based on position type secondarily', () => {
+      fakeGetBoundingClientRect.callsFake(() => {
+        return {
+          top: 250,
+          bottom: 250,
+        };
+      });
+      const positionPoints = constructPositionPoints(fakeAnchors);
+      for (let i = 0; i < fakeAnchors.length; i++) {
+        // The bottom position for all of the fake anchors is the same, so
+        // those points will all be at the top of the list
+        assert.equal(positionPoints.points[i][2], fakeAnchors[i]);
+        // This point is a "bottom" point
+        assert.equal(positionPoints.points[i][1], -1);
+        // The top position for all of the fake anchors is the same, so
+        // they'll be sorted to the end of the list
+        assert.equal(
+          positionPoints.points[i + fakeAnchors.length][2],
+          fakeAnchors[i]
+        );
+        // This point is a "top" point
+        assert.equal(positionPoints.points[i + fakeAnchors.length][1], 1);
+      }
+    });
+  });
+  describe('buildBuckets', () => {
```

Comment:
Can you add an empty line above this `describe` block.

Reply:
Yes I can.